pydicom/values.py

```diff
@@ -692,7 +787,16 @@ def convert_value(VR, raw_data_element, encodings=None):
 # the converter maps to a tuple
 # (function, struct_format)
 # (struct_format in python struct module style)
-converters = {
+_ConverterType = Dict[
+    str,
+    Union[
+        # Non-text and non-numeric VRs, Text VRs and PN, SQ
+        Callable[..., Union[object, SequenceType[object]]],
+        # Numeric VRs
+        Tuple[Callable[..., Union[object, SequenceType[object]]], str],
+    ]
+]
```

Comment:
Yikes - that must have taken a while to figure out.  I'm not going to try to confirm it aside from a cursory glance...

Reply:
This is the simple version! I originally had types for each signature and it was a mess. This still isn't perfect, either... `Callable` isn't great for compex signatures (I could use `Protocol` instead).